File: Issue-06/generate_sdf.py
import sys
import os

if __name__ == '__main__':
    args = sys.argv
    if len(args) not in [2,3]:
        print 'amount of parameters: 2'
        print 'usage: generate_sdf.py <inputfile> <delimiter>'
        exit(0)
    else:
        if len(args) == 2:
            args.append(' ')
        FNAME = sys.argv[1]
        delimiter = args[2]
        f = open(FNAME, 'r')
        if os.path.isfile(FNAME):
            s = 0
            for line in f:
                s += int(line.split(delimiter)[1])
            c = 0
            f.close()
            f = open(FNAME, 'r')
            for line in f:
                line = line.replace('\n','')
                p = int(line.split(delimiter)[1]) / (s * 1.0)
                c += p
                print '%s %f %f' %(line, p, c)

        else:
            print 'No such file or directory'

File: Issue-06/generate_sdf.m
#! /bin/octave -qf
args = argv();
if (length(args)>=1 & length(args) <= 2)
	delimiter = " ";
	if (length(args) == 2)
    delimiter = args{2};
	endif
	FNAME = args{1};
	fid = fopen(FNAME, "r");
	if(is_valid_file_id(fid))
		# First option
		# [A,B] = textread(FNAME, "%f %f\n");
		# c = sum(B);
		s = 0;
		while(!feof(fid))
			[A,B] = fscanf(fid,"%f %d","C");
			s = s + B;
		endwhile
		fclose(fid);
		fid = fopen(FNAME, "r");
		c = 0;
		while(!feof(fid))
			[A,B] = fscanf(fid,"%f %d","C");
			p = B / s;
			c = c + p;
			printf("%f %d %f %f\n", A, B, p, c);
		endwhile
		fclose(fid);
	else
		printf("No such file or directory\n");
	endif
else 
  printf("amount of parameters: 2\n");
  printf("usage: generate_sdf.py <inputfile> <delimiter>\n");
  exit;
endif